* The value for RIPEMD-160 in epg-digest-algorithm-alist has been fixed.
  Patch: fix-epg-digest-algorithm-alist.diff
  Provided-by: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
  Date: Thu, 26 Aug 2010 14:09:01 UTC
  Added-by: Rob Browning <rlb@defaultvalue.org>
  Status: incorporated upstream

  The Debian patch is taken from this upstream commit:

  revno: 102553
  committer: Daiki Ueno <ueno@unixuser.org>
  branch nick: trunk
  timestamp: Fri 2010-12-03 10:52:43 +0900
  message:
    Trivial fixes for epg.el.

    * epg.el (epg-digest-algorithm-alist): Replace "RMD160" with
    "RIPEMD160" (Bug#7490).  Reported by Daniel Kahn Gillmor.
    (epg-context-set-passphrase-callback): Mention that the callback
    is not called when used with GnuPG 2.x.

--- a/lisp/ChangeLog
+++ b/lisp/ChangeLog
@@ -1,3 +1,10 @@
+2010-12-03  Daiki Ueno  <ueno@unixuser.org>
+
+	* epg.el (epg-digest-algorithm-alist): Replace "RMD160" with
+	"RIPEMD160" (Bug#7490).  Reported by Daniel Kahn Gillmor.
+	(epg-context-set-passphrase-callback): Mention that the callback
+	is not called when used with GnuPG 2.x.
+
 2010-05-07  Chong Yidong  <cyd@stupidchicken.com>
 
 	* Version 23.2 released.
--- a/lisp/epg.el
+++ b/lisp/epg.el
@@ -66,7 +66,7 @@
 (defconst epg-digest-algorithm-alist
   '((1 . "MD5")
     (2 . "SHA1")
-    (3 . "RMD160")
+    (3 . "RIPEMD160")
     (8 . "SHA256")
     (9 . "SHA384")
     (10 . "SHA512")
@@ -335,7 +335,13 @@
 car is a function and cdr is a callback data.
 
 The function gets three arguments: the context, the key-id in
-question, and the callback data (if any)."
+question, and the callback data (if any).
+
+The callback may not be called if you use GnuPG 2.x, which relies
+on the external program called `gpg-agent' for passphrase query.
+If you really want to intercept passphrase query, consider
+installing GnuPG 1.x _along with_ GnuPG 2.x, which does passphrase
+query by itself and Emacs can intercept them."
   (unless (eq (car-safe context) 'epg-context)
     (signal 'wrong-type-argument (list 'epg-context-p context)))
   (aset (cdr context) 7 (if (consp passphrase-callback)
